cmd/juju/cloud/add.go

```diff
@@ -264,15 +264,15 @@ func queryName(
 	}
 }
 
-func queryCloudType(pollster *interact.Pollster) (string, error) {
+// cloudProviders returns the names of providers supported by add-cloud,
+// and also the names of those which are not supported.
+func cloudProviders() (providers []string, unsupported []string, _ error) {
```

Comment:
The function name could explain what it returns a bit better. Maybe "addableCloudProviders" (sounds a bit weird, but explains better...)?